### ISSUE TITLE
Issue Setting Default Value for BigDecimal 

### ISSUE THREAD

**Issue by harrychap** [avro-kotlin/avro4k#62]
I am trying to set the default for a big decimal field as follows: 

```
@AvroName("myclass")
@Serializable
data class myclass(
    val name: String,
    @Serializable(BigDecimalSerializer::class) @AvroDefault("0") val decimal: BigDecimal? = BigDecimal.ONE
)

val schema = Avro.default.schema(myclass.serializer())
println(schema.toString())
```
and it is producing this error. What is the correct way to do this? 
```
Exception in thread "main" org.apache.avro.AvroRuntimeException: Unknown datum class: class java.lang.Byte
	at org.apache.avro.util.internal.JacksonUtils.toJson(JacksonUtils.java:87)
	at org.apache.avro.util.internal.JacksonUtils.toJsonNode(JacksonUtils.java:48)
	at org.apache.avro.Schema$Field.<init>(Schema.java:558)
	at com.sksamuel.avro4k.schema.ClassSchemaFor.buildField(ClassSchemaFor.kt:103)
	at com.sksamuel.avro4k.schema.ClassSchemaFor.dataClassSchema(ClassSchemaFor.kt:37)
	at com.sksamuel.avro4k.schema.ClassSchemaFor.schema(ClassSchemaFor.kt:25)
	at com.sksamuel.avro4k.Avro.schema(Avro.kt:218)
```

**Comment by thake** [comment]
Should be fixed with PR #73 

However, it needs to be specified as a byte default the same way one would need to specify it in the avro schema. See https://avro.apache.org/docs/current/spec.html#schema_record for details.

For example a default value equivalent to 0 would be encoded as @AvroDefault("\u0000").

Can you check if this works now for you with the current master?

**Comment by thake** [comment]
Closing the issue for now. If you still experience problems with default values for BigDecimals, please reopen the issue.
